qiling/debugger/qdb/branch_predictor/branch_predictor_arm.py

```diff
@@ -40,9 +43,9 @@ def get_cpsr(bits: int) -> (bool, bool, bool, bool):
                 bits & 0x80000000 != 0, # N, sign flag
                 )
 
-    def predict(self):
+    def predict(self, pref_addr: Optional[int] = None) -> BranchPredictor.Prophecy:
         prophecy = self.Prophecy()
-        cur_addr = self.cur_addr
+        cur_addr = pref_addr or self.cur_addr
```

Comment:
Please note that the `or` operator will pick the latter not only when the former is `None`, but also when it is `0`. If `0` is a valid value for `pref_addr`, then consider using:

```python
cur_addr = self.cur_addr if pref_addr is None else pref_addr
```

Reply:
thx for the hint, i will correct this in the following fix for the issue

Reply:
@ucgJhe please update the code ?